libass/ass_drawing.c

```diff
@@ -43,108 +43,232 @@ static bool token_check_values(ASS_DrawingToken *token, int i, ASS_TokenType typ
     return true;
 }
 
+/*
+ * \brief Free a list of tokens
+ */
+static void drawing_free_tokens(ASS_DrawingToken *token)
+{
+    while (token) {
+        ASS_DrawingToken *at = token;
+        token = token->next;
+        free(at);
+    }
+}
+
+static inline bool add_node(ASS_DrawingToken **tail, ASS_TokenType type, ASS_Vector point)
+{
+    assert(tail && *tail);
+
+    (*tail)->next = malloc(sizeof(**tail));
+    ASS_DrawingToken *new_tail = (*tail)->next;
+    if (!new_tail)
+        return false;
+    new_tail->prev = *tail;
+    new_tail->next = NULL;
+    new_tail->type = type;
+    new_tail->point = point;
+    *tail = new_tail;
+    return true;
+}
+
+static inline bool get_point(const char **str, ASS_Vector *point)
+{
+    double x, y;
+    if (!mystrtod((char **) str, &x) || !mystrtod((char **) str, &y))
+        return false;
+    *point = (ASS_Vector) {double_to_d6(x), double_to_d6(y)};
+    return true;
+}
+
+
+/**
+ * Parses and advances the string for exactly 3 points.
+ * If the string does not contain fewer than 3 points,
```

Comment:
This is probably meant to say “contains fewer than 3 points” or “does not contain 3 points”

Reply:
fixed thx